From 77b27c07df2bdff86945766a4728b6012062535c Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan@agner.ch>
Date: Tue, 26 Nov 2019 09:20:30 +0100
Subject: [PATCH] Avoid race condition when building outside of source tree

When building outside of source tree it can happen that src/ostree/
does not exist (yet) when bison is called. This leads to an build
error like so:
  bison: src/ostree/parse-datetime.c: cannot open: No such file or directory

Make sure that src/ostree/ exists when parse-datetime.c is built.

Upstream-Status: Submitted
Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
---
 Makefile-ostree.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile-ostree.am b/Makefile-ostree.am
index 76f39cad..f861afe4 100644
--- a/Makefile-ostree.am
+++ b/Makefile-ostree.am
@@ -123,6 +123,7 @@ ostree_SOURCES += src/ostree/ot-remote-builtin-add-cookie.c \
 endif
 
 src/ostree/parse-datetime.c: src/ostree/parse-datetime.y Makefile
+	mkdir -p src/ostree/
 	$(AM_V_GEN) $(YACC) $< -o $@
 
 EXTRA_DIST += src/ostree/parse-datetime.y
-- 
2.24.0

